### ISSUE TITLE
Web links in menu won't open on Android

### ISSUE THREAD

**Issue by rbeesley** [zathras/jrpn#45]
**JRPN 16C**

**Help->About->"https://jron.jovial.com" or Help->Submit Issue (Web) result in net::ERR_CACHE_MISS error. Probably all such links will cause this error.**

**Debug Log**
N/A

**To Reproduce**
Help->About->"https://jron.jovial.com" or Help->Submit Issue (Web) result in net::ERR_CACHE_MISS error. Probably all such links will cause this error, but this was what I found before submitting this bug report.

![Screenshot_20230702-000845.png](https://github.com/zathras/jrpn/assets/3703112/fb61e3b8-b9c2-4fac-bb59-cd0879d28eb8)

**Expected behavior**
Web links should take the user to the expected web page. 

**Platform:**
 If relevant, some details on the platform are sometimes helpful.
- OS: Android 13 on Pixel Fold

**Additional context**

**Comment by zathras** [comment]
Thanks for reporting this.  It looks like I just need to request the network permissions:
```
<uses-permission android:name="android.permission.INTERNET" />
<uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
```
These are probably granted by default in debug mode.

**Comment by rbeesley** [comment]
Glad I could help.

**Comment by zathras** [comment]
Reproduced with release build option; just requesting INTERNET fixes it.

Unless requested otherwise, it'll probably be a few weeks before I spin a release for the Android store.  It would be nice to address #46.

**Comment by noahfriedman** [comment]
No. This solution is wrong wrong wrong wrong WRONG.

Wrong!

This application absolutely should not have internet permission and a soluteky should not be opening web links directly.  You should be using the Android intent mechanism to launch the user's preferred web browser!

There are a litany of reasons the current behavior is bad.  Security reasons (for example I'll be damned if I'm going to input my GitHub user credentials into a random android app's web view just to report a bug), as well as major UI considerations.

PLEASE revert this change.

**Comment by zathras** [comment]
@noahfriedman wrote:
> No. This solution is wrong wrong wrong wrong WRONG.
> 
> This application absolutely should not have internet permission and absolutely should not be opening web links directly. You should be using the Android intent mechanism to launch the user's preferred web browser as an external process.
> 
> There are a litany of reasons the current behavior is bad. Security reasons (for example I'll be damned if I'm going to input my GitHub user credentials into a random android app's web view just to report a bug), as well as major UI considerations—integration with password managers, forms, tabs, ability to switch between the actual calculator and the web form when trying to describe a bug's steps to reproduce accurately, and so on.
> 
> PLEASE revert this change.

Well, since you're displaying such a high level of maturity, I'll get right on it.

Your diagnosis is wrong, but you do have a point.  What actually happened is that Google changed the semantics of a Flutter library out from under me.  Reverting the change would be silly; it looks like getting the old behavior back is probably an Android-specific configuration thing.

I'm not quite sure where all this emotion is coming from, over a pretty marginal thing.  Do you really plan to submit that many github issues from your phone?  Bit of an odd kink, but OK.

**Comment by zathras** [comment]
Well, Noah is about 30% correct, but changing to a different way of rendering belongs in a different issue.  This issue (45) is fixed and released in 2.1.5.
